src/persistence/common.jl

```diff
@@ -44,7 +44,7 @@ function auto_decompress(io::IO)
         b1 = read(io, UInt8)
         if !eof(io)
             b2 = read(io, UInt8)
-            if (b1, b2) == (0x1f, 0x8b)  # check magic bytes
+            if b1 == 0x1f && b2 == 0x8b  # check magic bytes
```

Comment:
Is this the same issue with `missing` as above then?

Reply:
Yes but for some reason this tweak solved it :shrug:

Reply:
This is because the types of `b1` and `b2` is not inferred given the input argument type `io::IO`  (since there are too many matching methods for `read(::IO, ::Type{UInt8})`. Alternatively you can annotate `b1 = read(io, UInt8)::UInt8` instead (which I agree that looks silly though...)

Reply:
thanks @aviatesk! I think we'll keep it that way for now